timemachine/potentials/bonded.py

```diff
@@ -147,6 +148,88 @@ def harmonic_bond(conf, params, box, lamb, bond_idxs, lamb_mult=None, lamb_offse
     return jnp.sum(energy)
 
 
+def get_cos_angle(ci, cj, ck):
+
+    ci = jnp.array(ci)
+    cj = jnp.array(cj)
+    ck = jnp.array(ck)
+
+    # compute cosine angle of three points
+    vij = ci - cj
+    vjk = ck - cj
+    top = jnp.sum(jnp.multiply(vij, vjk), -1)
+    bot = jnp.linalg.norm(vij, axis=-1) * jnp.linalg.norm(vjk, axis=-1)
+
+    tb = top / bot
+
+    return tb
+
+
+def get_centroid_cos_angles(conf, angle_idxs):
+
+    x_is = []
+    x_js = []
+    x_ks = []
+
+    for idxs, j, k in angle_idxs:
+        x_j = conf[j]
+        # compute v-site from unit vectors to avoid bond length
+        # change differences
+        a, b = idxs
```

Comment:
Makes it slightly easier to skim implementation to remind what format to provide `angle_idxs`

```suggestion
    for (a, b), j, k in angle_idxs:
        x_j = conf[j]
        # compute v-site from unit vectors to avoid bond length
        # change differences
```

Reply:
I may modify this code a little bit so its more easily vectorizable in a future pass. `jax.jit` really doesn't like for loops.